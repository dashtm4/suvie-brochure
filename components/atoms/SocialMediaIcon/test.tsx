import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { SocialMediaIcon } from '.'

describe('<SocialMediaIcon />', () => {
  const socialMedia = {
    title: 'Facebook',
    url: 'https://www.facebook.com/SuvieFood/',
    icon: '/images/facebook-icon.svg'
  }

  it('should render correct a tag', () => {
    const wrapper = shallow(<SocialMediaIcon socialMedia={socialMedia} />)

    expect(wrapper.html()).toContain(
      `<a href="${socialMedia.url}" aria-label="${socialMedia.title} - opens in new window" target="_blank" rel="noreferrer"`
    )

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should render correct html with correct target attribute', () => {
    const target = '_parent'

    const wrapper = shallow(
      <SocialMediaIcon socialMedia={socialMedia} target={target} />
    )

    expect(wrapper.html()).toContain(
      `<a href="${socialMedia.url}" aria-label="${socialMedia.title}" target="${target}" rel="noreferrer"`
    )

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should render correct html with correct rel attribute', () => {
    const rel = 'help'

    const wrapper = shallow(
      <SocialMediaIcon socialMedia={socialMedia} rel={rel} />
    )

    expect(wrapper.html()).toContain(
      `<a href="${socialMedia.url}" aria-label="${socialMedia.title} - opens in new window" target="_blank" rel="${rel}"`
    )

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should contain img', () => {
    const wrapper = shallow(<SocialMediaIcon socialMedia={socialMedia} />)

    expect(wrapper.html()).toContain('<img')
  })
})
