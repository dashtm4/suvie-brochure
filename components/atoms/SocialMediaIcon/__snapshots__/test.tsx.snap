// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialMediaIcon /> should render correct a tag 1`] = `
<a
  aria-label="Facebook - opens in new window"
  href="https://www.facebook.com/SuvieFood/"
  rel="noreferrer"
  target="_blank"
>
  <Image
    alt="Facebook"
    height={30}
    src="/images/facebook-icon.svg"
    width={30}
  />
</a>
`;

exports[`<SocialMediaIcon /> should render correct html with correct rel attribute 1`] = `
<a
  aria-label="Facebook - opens in new window"
  href="https://www.facebook.com/SuvieFood/"
  rel="help"
  target="_blank"
>
  <Image
    alt="Facebook"
    height={30}
    src="/images/facebook-icon.svg"
    width={30}
  />
</a>
`;

exports[`<SocialMediaIcon /> should render correct html with correct target attribute 1`] = `
<a
  aria-label="Facebook"
  href="https://www.facebook.com/SuvieFood/"
  rel="noreferrer"
  target="_parent"
>
  <Image
    alt="Facebook"
    height={30}
    src="/images/facebook-icon.svg"
    width={30}
  />
</a>
`;
